ratpack-bearer-auth/src/main/java/st/ratpack/auth/springsec/jwt/JWTTokenValidator.java

```diff
@@ -0,0 +1,72 @@
+package st.ratpack.auth.springsec.jwt;
+
+import com.nimbusds.jose.JOSEException;
+import com.nimbusds.jose.crypto.RSASSAVerifier;
+import com.nimbusds.jose.jwk.JWK;
+import com.nimbusds.jose.jwk.RSAKey;
+import com.nimbusds.jwt.SignedJWT;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import ratpack.exec.Promise;
+import st.ratpack.auth.OAuthToken;
+import st.ratpack.auth.TokenValidator;
+
+import java.text.ParseException;
+import java.util.Date;
+import java.util.Optional;
+
+public class JWTTokenValidator implements TokenValidator {
+	private static Logger logger = LoggerFactory.getLogger(JWTTokenValidator.class);
+	private final JWKProvider jwkProvider;
+	private final TokenValidator upstreamTokenValidator;
+
+	private static final String EXPIRATION_TIME_CLAIM = "exp";
+	private static final String JWT_REGEX = "^[a-zA-Z0-9\\-_]+?\\.[a-zA-Z0-9\\-_]+?\\.([a-zA-Z0-9\\-_]+)?$";
+
+	public JWTTokenValidator(
+			TokenValidator upstreamValidator,
+			JWKProvider JWKSetProvider
+	) {
+		jwkProvider = JWKSetProvider;
+		upstreamTokenValidator = upstreamValidator;
+	}
+
+	@Override
+	public Promise<Optional<OAuthToken>> validate(String token) {
+		if (isJwtToken(token) && !verifyJwtToken(token)) {
```

Comment:
It's a good start I think it may make sense to allow for a predicate and separate JWT validation from the I'll spike something out